subsys/net/lib/nrf_cloud/src/nrf_cloud_alerts.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <date_time.h>
+#include "nrf_cloud_fsm.h"
+#include "nrf_cloud_mem.h"
+#include "nrf_cloud_codec.h"
+#include "nrf_cloud_transport.h"
+#include <net/nrf_cloud_rest.h>
+#include <net/nrf_cloud_alerts.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_alerts, CONFIG_NRF_CLOUD_ALERTS_LOG_LEVEL);
+
+static int alerts_enabled = IS_ENABLED(CONFIG_NRF_CLOUD_ALERTS);
+
+#if defined(CONFIG_NRF_CLOUD_ALERTS)
+static atomic_t alert_sequence;
+
+static int alert_prepare(struct nrf_cloud_data *output,
+			 enum nrf_cloud_alert_type type,
+			 float value,
+			 const char *description)
+{
+	struct nrf_cloud_alert_info alert = {.ts_ms = 0, .type = type, .value = value,
+					     .description = description};
+
+	if (IS_ENABLED(CONFIG_DATE_TIME)) {
+		date_time_now(&alert.ts_ms);
+	}
+
+	if (!alert.ts_ms || IS_ENABLED(CONFIG_NRF_CLOUD_ALERTS_SEQ_ALWAYS)) {
+		alert.sequence = ++alert_sequence;
+	}
+
+	LOG_DBG("Alert! type: %d, value: %f, description: %s, ts: %lld, seq: %d",
+		alert.type, alert.value, alert.description ? alert.description : "",
+		alert.ts_ms, alert.sequence);
+
+	return nrf_cloud_encode_alert(&alert, output);
+}
+#endif /* CONFIG_NRF_CLOUD_ALERTS */
+
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+int nrf_cloud_alert_send(enum nrf_cloud_alert_type type,
+			 float value,
+			 const char *description)
+{
+#if defined(CONFIG_NRF_CLOUD_ALERTS)
+	struct nct_dc_data output = {.message_id = NCT_MSG_ID_USE_NEXT_INCREMENT};
+	int err;
+
+	if (!alerts_enabled) {
+		return 0;
+	}
+
+	if (nfsm_get_current_state() != STATE_DC_CONNECTED) {
+		LOG_ERR("Cannot send alert; not connected");
+		return -EACCES;
+	}
+
+	err = alert_prepare(&output.data, type, value, description);
+	if (!err) {
+		LOG_DBG("Encoded alert: %s", (const char *)output.data.ptr);
+	} else {
+		LOG_ERR("Error encoding alert: %d", err);
+		return err;
+	}
+
+	err = nct_dc_send(&output);
```

Comment:
Would it be preferable to use `nrf_cloud_send` here?

Reply:
good point